File: Problems/Find-if-Path-Exists-in-Graph/Find-if-Path-Exists-in-Graph.java
class Solution {
    public boolean validPath(int n, int[][] edges, int source, int destination) {
        Set<Integer> visited = new HashSet<>();
        Deque<Integer> remaining = new ArrayDeque<>();

        visited.add(source);
        for (int[] edge : edges) {
            if (edge[0] == source && remaining.offer(edge[1]))
                visited.add(edge[1]);
            if (edge[1] == source && remaining.offer(edge[0]))
                visited.add(edge[0]);
        }
        while (!remaining.isEmpty() && !visited.contains(destination)) {
            int vertex = remaining.poll();
            for (int[] edge : edges) {
                if (edge[0] == vertex && !visited.contains(edge[1])) {
                    remaining.offer(edge[1]);
                    visited.add(edge[1]);
                }
                if (edge[1] == vertex && !visited.contains(edge[0])) {
                    remaining.offer(edge[0]);
                    visited.add(edge[0]);
                }
            }
        }

        return visited.contains(destination);
    }
}

File: Problems/Find-if-Path-Exists-in-Graph/Find-if-Path-Exists-in-Graph.js
/**
 * @param {number} n
 * @param {number[][]} edges
 * @param {number} source
 * @param {number} destination
 * @return {boolean}
 */
var validPath = function(n, edges, source, destination) {
    const visited = new Set([source]);
    const remaining = [];

    edges.forEach(edge => {
        if (edge[0] == source)
            remaining.push(edge[1]), visited.add(edge[1]);
        if (edge[1] == source)
            remaining.push(edge[0]), visited.add(edge[0]);
    });
    while (remaining.length && !visited.has(destination)) {
        const vertex = remaining.pop();
        edges.forEach(edge => {
            if (edge[0] == vertex && !visited.has(edge[1])) {
                remaining.push(edge[1]);
                visited.add(edge[1]);
            }
            if (edge[1] == vertex && !visited.has(edge[0])) {
                remaining.push(edge[0]);
                visited.add(edge[0]);
            }
        });
    }
    return visited.has(destination);
};
